frontend_tests/puppeteer_tests/recent-topic.ts

```diff
@@ -0,0 +1,446 @@
+        "database isn't skipping erratically",
+        "plotter",
+        "green printer is running quickly",
+    ]);
+}
+
+// Marks a specific topic as read.
+async function mark_as_read(page: Page, topic: string): Promise<void> {
+    await page.evaluate((topic: string) => {
+        const mark_as_read_icon = $(`.on_hover_topic_read[data-topic-name='${topic}']`);
+        mark_as_read_icon.trigger("click");
+    }, topic);
+}
+
+// Mutes the topic for the corresponding topic name passed.
+async function mute_topic(page: Page, topic: string): Promise<void> {
+    await page.evaluate((topic: string) => {
+        const mute_icon = $(`.on_hover_topic_mute[data-topic-name='${topic}']`);
+        mute_icon.trigger("click");
+    }, topic);
+}
+
+// Should yield same result as `test_initial_topics`.
+async function test_all_filter_button(page: Page): Promise<void> {
+    await page.waitForSelector(recent_topic_page);
+
+    // Triggering the `All` filter button.
+    await trigger_filter_button(page, all_filter_button);
+
+    const stream_names = await get_stream_names(page);
+    assert.deepStrictEqual(stream_names, [
+        "Verona",
+        "Denmark",
+        "Venice",
+        "Denmark",
+        "Venice",
+        "Verona",
+        "Verona",
+    ]);
+
+    const topic_names = await get_topic_names(page);
+    assert.deepStrictEqual(topic_names, [
+        "last nas is loading",
+        "last database linking",
+        "URLS",
+        "last server wasn't loading slowly",
+        "database isn't skipping erratically",
+        "plotter",
+        "green printer is running quickly",
+    ]);
+}
+
+async function test_include_muted_filter_button(page: Page): Promise<void> {
+    await page.waitForSelector(recent_topic_page);
+
+    // Triggering the `Include muted` filter button.
+    await trigger_filter_button(page, include_muted_filter_button);
+
+    const stream_names = await get_stream_names(page);
+    assert.deepStrictEqual(stream_names, [
+        "Verona",
+        "Denmark",
+        "Venice",
+        "Denmark",
+        "Venice",
+        "Verona",
+        "Verona",
+    ]);
+
+    const topic_names = await get_topic_names(page);
+    assert.deepStrictEqual(topic_names, [
+        "last nas is loading",
+        "last database linking",
+        "URLS",
+        "last server wasn't loading slowly",
+        "database isn't skipping erratically",
+        "plotter",
+        "green printer is running quickly",
+    ]);
+
+    // Unchecking the checkbox by clicking the button again.
+    await trigger_filter_button(page, include_muted_filter_button);
+}
+
+async function test_unread_filter_button(page: Page): Promise<void> {
+    await page.waitForSelector(recent_topic_page);
+
+    // Triggering the `Unread` filter button.
+    await trigger_filter_button(page, unread_filter_button);
+
+    const stream_names = await get_stream_names(page);
+    assert.deepStrictEqual(stream_names, []);
```

Comment:
By default there isn't any unread topic/muted topic when the puppeteer logs in. This leads to `Unread` filter button returns a empty array of topics and `Include Muted` filter button to return the same result as `All` filter button.
(Mentioned this as I somehow felt like this isn't the right way to proceed.)